driver/format/RowBinaryWithNamesAndTypes.cpp

```diff
@@ -152,14 +153,24 @@ void RowBinaryWithNamesAndTypesResultSet::readValue(WireTypeDateTimeAsInt & dest
     readPOD(dest.value);
 }
 
+void RowBinaryWithNamesAndTypesResultSet::readValue(WireTypeDateTime64AsInt & dest, ColumnInfo & column_info) {
+    readPOD(dest.value);
+}
+
 void RowBinaryWithNamesAndTypesResultSet::readValue(DataSourceType<DataSourceTypeId::Date> & dest, ColumnInfo & column_info) {
-    WireTypeDateAsInt dest_raw;
+    WireTypeDateAsInt dest_raw(column_info.timezone);
```

Comment:
`Date` has no timezone in CH

Reply:
Yeah, but the timezone (of the data/server) might be used when it is converted (read by the user) to some DateTime structures.